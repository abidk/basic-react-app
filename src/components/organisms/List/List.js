import React from 'react';
import PropTypes from 'prop-types';
import ListItem from '../../molecules/ListItem';
import './List.css';

const List = ({ names }) => <ul>{names.map(name => <ListItem key={name} name={name} />)}</ul>

List.propTypes = {
    names: PropTypes.arrayOf(PropTypes.string).isRequired
}

export default List;