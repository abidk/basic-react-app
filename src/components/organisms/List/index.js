import { connect } from "react-redux";
import List from './List';

const mapStateToProps = (state, props) => {
  return { 
    names: [...props.names, ...state.names]
  };
};

export default connect(mapStateToProps)(List);