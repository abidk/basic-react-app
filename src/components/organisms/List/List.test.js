import React from 'react'
import configureMockStore from 'redux-mock-store'
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import List from './index';

const mockStore = configureMockStore();

describe(__filename, () => {

    it('Should render', () => {    
        const store = mockStore({names:['Abid3']});

        const output = mount(<List names={['Abid1', 'Abid2']} store={store}/>);

        expect(shallowToJson(output)).toMatchSnapshot();
    });
});