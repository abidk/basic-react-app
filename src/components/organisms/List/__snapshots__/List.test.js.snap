// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/abidkhalil/Documents/basic-react-app/src/components/organisms/List/List.test.js Should render 1`] = `
<ConnectFunction
  names={
    Array [
      "Abid1",
      "Abid2",
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <List
    dispatch={[Function]}
    names={
      Array [
        "Abid1",
        "Abid2",
        "Abid3",
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <ul>
      <ListItem
        key="Abid1"
        name="Abid1"
      >
        <li
          className="ListItem"
        >
          <Hello
            isVisible={true}
            name="Abid1"
          >
            <p
              className="HelloText"
            >
              Hello 
              Abid1
            </p>
          </Hello>
          <Button
            action={[Function]}
            label="Show/Hide Abid1"
          >
            <button
              className="Button"
              onClick={[Function]}
            >
              Show/Hide Abid1
            </button>
          </Button>
        </li>
      </ListItem>
      <ListItem
        key="Abid2"
        name="Abid2"
      >
        <li
          className="ListItem"
        >
          <Hello
            isVisible={true}
            name="Abid2"
          >
            <p
              className="HelloText"
            >
              Hello 
              Abid2
            </p>
          </Hello>
          <Button
            action={[Function]}
            label="Show/Hide Abid2"
          >
            <button
              className="Button"
              onClick={[Function]}
            >
              Show/Hide Abid2
            </button>
          </Button>
        </li>
      </ListItem>
      <ListItem
        key="Abid3"
        name="Abid3"
      >
        <li
          className="ListItem"
        >
          <Hello
            isVisible={true}
            name="Abid3"
          >
            <p
              className="HelloText"
            >
              Hello 
              Abid3
            </p>
          </Hello>
          <Button
            action={[Function]}
            label="Show/Hide Abid3"
          >
            <button
              className="Button"
              onClick={[Function]}
            >
              Show/Hide Abid3
            </button>
          </Button>
        </li>
      </ListItem>
    </ul>
  </List>
</ConnectFunction>
`;
